sodym/trade.py

```diff
@@ -0,0 +1,79 @@
+import sys
+from pydantic import BaseModel as PydanticBaseModel
+from pydantic import model_validator
+from abc import abstractmethod
+from .named_dim_arrays import Parameter
+from .dimensions import DimensionSet
+
+class Trade(PydanticBaseModel):
+    """ A TradeModule handles the storing and calculation of trade data for a given MFASystem."""
+
+    dims : DimensionSet
+    imports : Parameter
+    exports : Parameter
+
+    @model_validator(mode='after')
+    def validate_region_dimension(self):
+        assert 'r' in self.imports.dims.letters, "Imports must have a Region dimension."
+        assert 'r' in self.exports.dims.letters, "Exports must have a Region dimension."
+
+        return self
+
+    def __getitem__(self, key):
+        if key == 'Imports':
+            return self.imports
+        elif key == 'Exports':
+            return self.exports
+        else:
+            raise KeyError(f"Key {key} not found in Trade data - has to be either 'Imports' or 'Exports'.")
+
+    def __setitem__(self, key, value):
+        if key == 'Imports':
+            self.imports = value
+        elif key == 'Exports':
+            self.exports = value
+        else:
+            raise KeyError(f"Key {key} has to be either 'Imports' or 'Exports'.")
```

Comment:
Hmm, I disagree, though it's just a taste thing: This is very nice in simson now as when assigning the parameters to the flow it looks like this 'flw['sysenv => eol_market']['Fe'][...]          = trd['Scrap']['Imports']' which is just much more in line with the rest of the syntax in the compute flows (rather than flw['sysenv => eol_market']['Fe'][...]          = trd['Scrap'].imports). Guess it doesn't matter much either way. What do you guys think?

Reply:
Okay, let's keep it. I prefer the dot notation, especially since your IDE should autocomplete this (i.e. it's harder to ask for something that doesn't exist), but this option is of course still available :)